### ISSUE TITLE
Bug: sam cli does not throw error and exit when --config-file does not exist

### ISSUE THREAD

**Issue by edify42** [aws/aws-sam-cli#3899]
### Description:
Hello maintainers 👋 

Currently when deploying via `sam deploy --config-file <file>`, if `<file>` does not exist sam ignores that without throwing an error.

While it's unlikely this has caused issues for anyone (as the stack-name being missing throws an error), it does make the debugging experience strange for the first time you see the error and not realise that you might have passed in the wrong location to the config file.

### Steps to reproduce:
Run the below anywhere:

```bash
sam deploy --config-file file-that-does-not-exist.toml
```

### Observed result:
```bash
Usage: sam deploy [OPTIONS]
Try 'sam deploy -h' for help.

Error: Missing option '--stack-name', 'sam deploy --guided' can be used to provide and save needed parameters for future deploys.
```

### Expected result:
I'd expect the error be thrown when trying to read / validate the toml file

```bash
Usage: sam deploy [OPTIONS]
Try 'sam deploy -h' for help.

Error: File <file> could not be found

OR

Error: File <file> could not be read as permissions on the file/directory prevent it being read by the current user
```

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: macos / linux
2. `sam --version`: SAM CLI, version 1.46.0
3. AWS region: ap-southeast-2

**Comment by jlgoolsbee** [comment]
I'm unsure if I should open a new bug or just comment on this one; I left a comment on #4044 but I'm also unsure if that's "visible" enough:

I believe this change has broken deploys that use a relative path to resolve a custom location for `samconfig.toml`; passing an absolute path works – so there is a workaround – but it isn't optimal.

Now busted (worked previously):

```
$ sam deploy --template-file "$app_name/template.yaml" --config-file ../samconfig.toml --config-env $env
Error: Config file ../samconfig.toml does not exist or could not be read
```

... but works when full path is provided:

```
$ sam deploy --template-file "$app_name/template.yaml" --config-file /full/path/to/samconfig.toml --config-env $env
File with same data already exists at [...], skipping upload

	Deploying with following values
	===============================
... etc.
```

**Comment by mildaniel** [comment]
A fix has been released in version 1.56.0 on 09/01. Please re-open if it persists.
